### ISSUE TITLE
DAML Engine does not check that key maintainers are signatories

### ISSUE THREAD

**Issue by hurryabit** [digital-asset/daml#1123]
The following scenario passes even though it shouldn't:
```haskell
template Foo with
    p : Party
    q : Party
  where
    signatory p
    key (p, q) : (Party, Party)
    maintainer q

test = scenario do
  p <- getParty "P"
  q <- getParty "Q"
  submit p do
    create Foo with ..
```

**Comment by hurryabit** [comment]
https://github.com/digital-asset/daml/blob/ed431b47171bca1b7efeda63933fd7f89a87e924/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/types/Ledger.scala#L702
The reasoning at this point is related to this issue.

**Comment by gerolf-da** [comment]
This should have been fixed by #1124.

**Comment by stefanobaghino-da** [comment]
> This should have been fixed by #1124.

Sounds about right. Can we close this?
